Update to Clang-9 and use CMake find_package

planetmarshall opened the pull request:
* Update to Clang Tooling 9 API
* Use CMake `find_package(Clang)` and targets on Linux to find Clang Tooling libraries
* Update readme

bgeller commented:
This was an item on our todo list however we had not started the work. Wonderful to have your pull request. Did you by any chance try clang 10 as it moved to stable a few months ago.  Would you consider updating the PR?

planetmarshall commented:
> Did you by any chance try clang 10 as it moved to stable a few months ago. Would you consider updating the PR?

Sure, I'll take a look later today.

planetmarshall commented:
Clang-10 works with no modifications to the PR

Unfortunately I haven't been able to test a Windows build due to compilation issues (missing `unistd.h`, `popen` etc - are you using MINGW to build on Windows?)

I've done some simplification of the CMake file but I haven't been able to test on all platforms.

planetmarshall commented:
Ok I've updated the CMake file and added some configuration code to support building with VS2019. Tested this locally on the test_build folder and seems to work for me, though you may want to make changes for your own build systems. I don't know what kind of setup you use.

bgeller commented:
We need RTTI enabled for clang.  In prior versions this was not turned on.  This is why we released our own prebuilt versions of clang and were unable to use any of the prebuilt packages. Can you verify if this has changed for all of our supported platforms?  

On Windows we build for MSVC and MinGW.  We support Windows MinGW 32 and 64 bit.

Arch, Centos8, FreeBSD 12
Debian 9, Debian 10, Fedora 28, 29, 30, 31, Ubuntu 16.04, 18.04, 20.04
OS X 10.14

Thanks for your help.

bgeller commented:
Your changes demonstrated the need that we should not include unistd.h in the LEX files. We found an %option command which will do this automatically and set a #ifdef so the header is not included. The option was added to the LEX source files. Generated LEX files should not be changed by hand so all changes were made to the source files.

Removed generated LEX isAtty() function by adding %option never-interactive in the src/parser files.

We are currently reviewing the remainder of your suggestions. Thanks so much for your testing and submission.

Barbara
